### PR TITLE
chore: adding index for new action

### LINKED ISSUE #23273: [Task]:  Add index to NewAction collection for PluginType
### Is there an existing issue for this?

- [X] I have searched the existing issues

## Description

The repository query has been changed to use the pluginType in the query as well, In order to speed-up the query, an index for pluginType needs to be placed on newAction collection

- Create index on pluginType for newAction

### PR DESCRIPTION
**Pull request by sondermanish**
#### Adding compound index in NewAction collection to improve getAction query performance. The index has been added on
-  ApplicationId
-  PluginType
-  deletedAt

Fixes #23273 

#### Type of change

- Chore (housekeeping or task changes that don't impact user perception)

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration011AddPluginTypeIndexToNewActionCollection.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import io.mongock.api.annotations.Execution;
+import io.mongock.api.annotations.RollbackExecution;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.index.Index;
+
+import static com.appsmith.server.migrations.DatabaseChangelog1.ensureIndexes;
+import static com.appsmith.server.migrations.DatabaseChangelog1.makeIndex;
+import static com.appsmith.server.repositories.ce.BaseAppsmithRepositoryCEImpl.fieldName;
+
+@ChangeUnit(order = "011", id = "index-for-new-action-collection", author = " ")
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration011AddPluginTypeIndexToNewActionCollection.java:15]: Can you name this migration better so it is easy to distinguish?

Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration011AddPluginTypeIndexToNewActionCollection.java]: changed to "application-id-plugin-type-deleted-at-compound-index-for-new-action"

**Review by nidhi-nair — APPROVED** [review]

**Comment by sondermanish** [comment]
skipping QA as this would be covered in regression

**Pull request merged**